### PR TITLE
Feature/ab2d-1862-or-condition

### PR DESCRIPTION
**Pull request by adaykin**
### Change Details

This is an RFC for the OR condition that has been discussed in various DASG eng. meetings

### Acceptance Validation

Review the technical approach and see if it will fit within API guidelines, as well as if this is feasible by BFD

### Feedback Requested

Whether or not it's the best technical approach to this problem, if proposed solution looks clean

### External References

- [BLUEBUTTON-xxx](https://jira.cms.gov/browse/BLUEBUTTON-xxx)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: rfcs/0006-or-condition.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Future Possibilities](#future-possibilities)
+
+## Motivation
+[Motivation]: #motivation
+
+AB2D has long running export jobs where the amount of time the job takes is critical for customer happiness. In order to
+lessen the amount of time a job takes to run, we will need to be able to improve performance in every section of our code.
+One part that takes a long time is the call to the Patient URL that maps contracts to beneficiaries on a monthly basis.
+Being able to query multiple months in the same call will reduce the load on the BFD database since it will reduce the
```
Comment (jzulim) [inline rfcs/0006-or-condition.md:32]: We should include proving this assumption with performance tests during the implementation phase.

File: rfcs/0006-or-condition.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+can provide an OR clause when necessary.
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+Before a solution goes into place, how can we validate that this actually does help performance?
+Performance testing this with real world examples, e.g. comparing AB2D performance querying 6 months separately
+vs. 1 query with 6 months.
+
+Will other groups use this solution?
```
Comment (jzulim) [inline rfcs/0006-or-condition.md:80]: @msnook @dtisza1 @DeirdreHolub - Does BB2, BCDA or DPC have a use case for this OR in regards to Part D Contracts?

File: rfcs/0006-or-condition.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+that multiple months can be included, e.g. https://bluebutton.cms.gov/resources/variables/ptdcntrct01-02-03
+
+This value can be parsed by BFD to determine if there should be an or parameter in the event there's a list of items
+sent over. One month would mean that there's no need for a month parameter. The SQL query in queryBeneficiaryIds
+can provide an OR clause when necessary.
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+Before a solution goes into place, how can we validate that this actually does help performance?
```
Comment (jzulim) [inline rfcs/0006-or-condition.md:76]: We can deploy a branch to our test environment and run performance tests manually?

File: rfcs/0006-or-condition.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+[Proposed Solution: Notable Alternatives]: #proposed-solution-notable-alternatives
+
+We have to make a design fit within the FHIR spec, so unfortunately there's not too much room for alternatives.
+This solution tries to make as few changes as possible to the code on BFD's side so that it's a simple one that
+can utilize the existing system for using a month parameter, and simply expand it to optionally use additional months.
+
+## Future Possibilities
+[Future Possibilities]: #future-possibilities
+
+There could also be an "or" condition for parameters beyond months, but those would come on a case by case basis.
```
Comment (jzulim) [inline rfcs/0006-or-condition.md:100]: 👍

**Review by jzulim — APPROVED** [review]
This looks like a reasonable addition. I've noted that we should proof / performance test this in our TEST environment with a branch / PR deploy.

File: rfcs/0006-or-condition.md (hunk 1 above)
Comment (adaykin) [reply rfcs/0006-or-condition.md:32]: Comparing this implementation to the master branch will be a good validation test I believe.

**Review by dshekhar18 — APPROVED** [review]
@adaykin Please deploy on BFD-TEST through a PR/Branch deployment. And compare it with SBX (without OR condition) response time and share the performance comp.

**Pull request merged**